### ISSUE TITLE
[ios, core] Images used in linePattern are rendered incorrectly.

### ISSUE THREAD

**Issue by fabian-guerra** [mapbox/mapbox-gl-native#15472]
According to the lineStyleLayer [doc](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/darwin/src/MGLLineStyleLayer.h#L490) images passed as `linePattern` should be a factor of 2 width for seamless render.

Passing a `16x16` px image for lower zoom levels and `32x32` px for higher zoom levels renders the image incorrectly.

### Steps to reproduce

 1. Add a line style layer.
```
CLLocationCoordinate2D lineCoordinates[4] = {
        CLLocationCoordinate2DMake(47.6025, -122.3327),
        CLLocationCoordinate2DMake(45.5189, -122.6726),
        CLLocationCoordinate2DMake(37.7790, -122.4177),
        CLLocationCoordinate2DMake(34.0532, -118.2349)
    };
MGLShapeSource *source = [[MGLShapeSource alloc] initWithIdentifier:@"test" shape:nil options:nil];
    MGLPolylineFeature *polylineFeature = [MGLPolylineFeature polylineWithCoordinates:lineCoordinates count:4];
    source.shape = polylineFeature;
    [self.mapView.style addSource:source];
    [self.mapView.style setImage:[UIImage imageNamed:@"lowresolution"] forName:@"lowresolution"];
    [self.mapView.style setImage:[UIImage imageNamed:@"highresolution"] forName:@"highresolution"];
MGLLineStyleLayer *lineStyleLayer = [[MGLLineStyleLayer alloc] initWithIdentifier:@"testterLayer" source:source];
    NSDictionary *stops = @{ @16 : [NSExpression expressionForConstantValue:@"highresolution"] };
    lineStyleLayer.linePattern = [NSExpression mgl_expressionForSteppingExpression:[NSExpression expressionForVariable:@"zoomLevel"]
                                                                    fromExpression:[NSExpression expressionForConstantValue:@"lowresolution"]
                                                                             stops:[NSExpression expressionForConstantValue:stops] ];
    lineStyleLayer.lineCap = [NSExpression expressionForConstantValue:[NSValue valueWithMGLLineCap:MGLLineCapRound]];
    lineStyleLayer.lineJoin = [NSExpression expressionForConstantValue:[NSValue valueWithMGLLineJoin:MGLLineJoinRound]];
    lineStyleLayer.lineWidth = [NSExpression expressionWithFormat:@"mgl_interpolate:withCurveType:parameters:stops:($zoomLevel, 'linear', nil, %@)", @{ @1: @8, @20: @8 }];
    [self.mapView.style addLayer:lineStyleLayer];
```

 2.
 3.

### Expected behavior

The image will render correctly
![sliced](https://user-images.githubusercontent.com/2745166/63619048-7fbd5880-c5a2-11e9-9f42-3d1ef8318512.png)

### Actual behavior

This is how is rendered in lower zoom levels.

![lowresolution](https://user-images.githubusercontent.com/2745166/63619152-b4c9ab00-c5a2-11e9-87c1-ebaf9d72a682.png)

At higher zoom level.

![highresolution](https://user-images.githubusercontent.com/2745166/63619199-d165e300-c5a2-11e9-837b-17c0fd1b9f1c.png)

### Configuration

**Mapbox SDK versions:**  `master`
**iOS/macOS versions:** 12.4
**Device/simulator models:** Simulator iPhone 8
**Xcode version:** 10.3

**Comment by karimnaaji** [comment]
Some before/after comparisons video in addition to the screenshots from taken from https://github.com/mapbox/mapbox-gl-js/pull/9266.

Taken from the branch https://github.com/mapbox/mapbox-gl-native/tree/fix-line-pattern in the glfw app, the branch has the potential fix from https://github.com/mapbox/mapbox-gl-js/pull/9266.
- before: [before.mov.zip](https://github.com/mapbox/mapbox-gl-native/files/4227477/before.mov.zip)
- after: [after.mov.zip](https://github.com/mapbox/mapbox-gl-native/files/4227476/after.mov.zip)

cc @chloekraw

**Comment by chloekraw** [comment]
Will be fixed by https://github.com/mapbox/mapbox-gl-native/issues/16359 -- closing in favour of that ticket!

**Comment by jaericson** [comment]
@karimnaaji @tobrun @1ec5 @tmpsantos, can you please provide an update as to whether https://github.com/mapbox/mapbox-gl-native/issues/16359 was ever fixed or not? It's linked but was closed as stale. Thank you!
